cfme/tests/services/test_rest_services.py

```diff
@@ -2055,3 +2057,186 @@ def test_delete_cart_from_collection(self, cart):
             tags: service
         """
         delete_resources_from_collection([cart])
+
+
+@pytest.fixture(scope="module")
+def set_run_automate_method(appliance):
+    reset_setting = appliance.advanced_settings["product"][
+        "run_automate_methods_on_service_api_submit"
+    ]
+    appliance.update_advanced_settings(
+        {"product": {"run_automate_methods_on_service_api_submit": True}}
+    )
+    yield
+    appliance.update_advanced_settings(
+        {"product": {"run_automate_methods_on_service_api_submit": reset_setting}}
+    )
+
+
+@pytest.fixture(scope="module")
+def automate_env_setup(klass):
+    # this script sets default value for the dialog element
+    script = """
+    dialog_field = $evm.object
+    dialog_field["sort_by"] = "value"
+    dialog_field["sort_order"] = "ascending"
+    dialog_field["data_type"] = "integer"
+    dialog_field["required"] = "true"
+    dialog_field["default_value"] = 7
+    dialog_field["values"] = {1 => "one", 2 => "two", 10 => "ten", 7 => "seven", 50 => "fifty"}
+    """
+    method = klass.methods.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        location="inline",
+        script=script,
+    )
+
+    klass.schema.add_fields({"name": "meth", "type": "Method", "data_type": "Integer"})
+
+    instance = klass.instances.create(
```

Comment:
Question:
Why do you need ```data_type``` as ```Integer``` here(Actually don't need to mention ```data_type```. By default it takes ```string```)?

Reply:
It throws error afair if you pass Integer instead of string.